src/generate-oas.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from "path";
+import fs from "fs-extra";
+import { generateFromOas, download } from "@commerce-apps/raml-toolkit";
+import Handlebars from "handlebars";
+
+type ApiSpecDetail = {
+  filepath: string;
+  filename: string;
+  name: string;
+  apiName: string;
+  directoryName: string;
+};
+
+const API_DIRECTORY = path.resolve(`${__dirname}/../apis`);
+const STATIC_DIRECTORY = path.join(__dirname, "./static");
+const TARGET_DIRECTORY = path.resolve(`${__dirname}/../renderedTemplates`);
+const TEMPLATE_DIRECTORY = path.resolve(`${__dirname}/../templatesOas`);
+const INDEX_TEMPLATE_LOCATION = path.resolve(
+  `${__dirname}/../templates/index.ts.hbs`
+);
+
+function kebabToCamelCase(str: string): string {
+  return str.replace(/-([a-z])/g, (match, letter: string) =>
+    letter.toUpperCase()
+  );
+}
+
+/**
+ * Helper function. Also contains explicit workaround for some API names.
+ */
+export function resolveApiName(name: string): string {
```

Comment:
Do we not need to handle the cases for Shopper Context, COR Preferences, and Catalogs? Or is that being handled somewhere else?

Reply:
Good catch. This handles the naming for the directories + the contents of the top level index.js

We want the top level index.js to import the namespace (ie. ShopperContexts) rather than the type (ShopperContext). I'll add the cases

Reply:
This function now explicitly handles ShopperContext, CORSPreferences, and Catalogs.